tests/MavenTests/testMzAligner.cpp

```diff
@@ -44,15 +35,65 @@ void TestMzAligner::testDoAlignment() {
     QVERIFY(true);
 }
 
-void TestMzAligner::testSaveFit(){
+void TestMzAligner::testObiWarp()
+{
+
+    MavenParameters* mavenparameters = new MavenParameters;
+    mavenparameters->compoundMassCutoffWindow->setMassCutoffAndType(5,"ppm");
+    mavenparameters->minQuality = 0.8;
+    mavenparameters->minIntensity = 10000;
+    mavenparameters->matchRtFlag = false;
+    mavenparameters->samples = samples->alignmentSamples;
+    mavenparameters->clsf = new ClassifierNeuralNet;
+    mavenparameters->clsf->loadModel("bin/default.model");
+
+    ObiParams params("cor", false, 2.0, 1.0, 0.20, 3.40, 0.0, 20.0, false, 0.60);
+    Aligner aligner;
+    aligner.alignWithObiWarp(mavenparameters->samples, &params);
+
+    DBS.loadCompoundCSVFile("bin/methods/KNOWNS.csv");
+    vector<Compound*> compounds = DBS.getCopoundsSubset("KNOWNS");
+
+
+    PeakDetector peakDetector;
+    peakDetector.setMavenParameters(mavenparameters);
+    vector<mzSlice*> slices = peakDetector.processCompounds(compounds, "compounds");
+    peakDetector.processSlices(slices, "compounds");
 
-    vector<mzSample*> samplesToLoad;
-    for (int i = 0; i <  files.size(); ++i) {
-        mzSample* sample = new mzSample();
-        sample->loadSample(files.at(i).toLatin1().data());
-        samplesToLoad.push_back(sample);
+    for(PeakGroup grp: mavenparameters->allgroups) {
+
+        if(grp.getName() == "NAD+" || grp.getName() == "methionine" || grp.getName() == "dTMP") {
+            float refSampleOriginalRt;
+            vector<float> sampleOriginalRt;
+            vector<float> sampleNewRt;
+            for(Peak peak: grp.peaks) {
+                if(peak.getSample()->sampleName == aligner.refSample->sampleName) {
+                    refSampleOriginalRt = peak.getScan()->originalRt;
+                }
+                else {
+                    sampleOriginalRt.push_back(peak.getScan()->originalRt);
+                    sampleNewRt.push_back(peak.getScan()->rt);
+                }
+            }
+            for(int rtCount=0; rtCount<sampleOriginalRt.size(); rtCount++) {
+                 float newDiff = std::abs(refSampleOriginalRt - sampleNewRt[rtCount]);
+                 float originalDiff = std::abs(refSampleOriginalRt - sampleOriginalRt[rtCount]);
+                 if(newDiff <= originalDiff)
+                     QVERIFY(true);
```

Comment:
Most assertion checkers can take in any expression that eventually evaluates to a single boolean, and if thats the case with `QVERIFY` then I think this set of checks can be be simplified to:
```C++
QVERIFY(newDiff <= originalDiff
        || (int)(newDiff) <= (int)(originalDiff))
```
I have not tried this myself though.